pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1019,6 +1019,17 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 			return
 		}
 
+		// setup workloadSds configs
+		sdsWorkloadConfig, err := authn_model.NewWorkloadSdsConfigBuilder().
```

Comment:
IMO keep originally code(use meshconfig) is simpler (no need to use java? style here)

Reply:
True that this adds a bit of verbosity. The goal was to provide a clear place to validate the workload SDS configuration (`Build()` step), but this might be better with just a regular struct and a validation helper?